### DISCUSSION TITLE
correct way to handle setting custom `rootPath` on an initial install?

### DISCUSSION THREAD

**Discussion by maphew** [ScoopInstaller/Scoop#4622] [Help and Questions]
What's the correct way to handle setting custom `rootPath` on an initial install?

~~~
Write-Output "--- Setting up env-gis-dev on a new machine or user profile ---"
Write-Output "---- Allowing powershell scripts..."
Set-ExecutionPolicy RemoteSigned -scope CurrentUser
if (-Not (Get-Command scoop -ErrorAction SilentlyContinue)) {
    echo "---- Installing Scoop command line program installer..."
    Invoke-Expression (New-Object System.Net.WebClient).DownloadString('https://get.scoop.sh')
    }
Write-Output "---- Configure scoop defaults"
scoop config cachePath \\server\share\scoop\cache
scoop config rootPath C:\apps\geo
~~~

Results in:
~~~
--- Setting up env-gis-dev on a new machine or user profile ---                                                                                      ---- Allowing powershell scripts...                                                                                                                  ---- Configure scoop defaults
Get-ChildItem : Cannot find path 'C:\apps\geo\shims' because it does not exist.
At C:\Users\matt\scoop\apps\scoop\current\lib\commands.ps1:3 char:12
+         + (Get-ChildItem "$scoopdir\shims") `
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\apps\geo\shims:String) [Get-ChildItem], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

'cachePath' has been set to '\\server\share\scoop\cache'
Get-ChildItem : Cannot find path 'C:\apps\geo\shims' because it does not exist.
At C:\Users\matt\scoop\apps\scoop\current\lib\commands.ps1:3 char:12
+         + (Get-ChildItem "$scoopdir\shims") `
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\apps\geo\shims:String) [Get-ChildItem], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

'rootPath' has been set to 'C:\apps\geo'
~~~

**Comment by maphew** [comment]
creating `$rootPath\shims` ahead of setting the config seems to work:

~~~
$rootPath = "c:\apps\geo"
$dirs = "shims"
foreach ($d in $dirs) {
    if (-Not (Get-Item $rootPath\$d -ErrorAction SilentlyContinue)) {
        New-Item $rootPath\$d -ItemType Directory
    }}
scoop config rootPath $rootPath
~~~

**Reply by rashil2000** [reply]
Seems like it. It should probably be documented somewhere.

**Comment by maphew** [comment]
I looked for docs on rootPath  everywhere I could think of without luck.
Maybe scoop should create the path if it doesn't exist?

**Reply by rashil2000** [reply]
Yes. Can you create a PR for the bug?

**Comment by niheaven** [comment]
It's very strange here, since this line: https://github.com/ScoopInstaller/Scoop/blob/5602083868ed04224ea8edf1ff0c3a2a3f6588b9/bin/install.ps1#L56 will call `shim()` which will create shim folder.

`rootPath`'s help is at the bottom of `lib\core.ps1`:

https://github.com/ScoopInstaller/Scoop/blob/5602083868ed04224ea8edf1ff0c3a2a3f6588b9/lib/core.ps1#L1024-L1043

**Reply by niheaven** [reply]
In your output, I haven't see the output of `Write-Output "---- Allowing powershell scripts..."`, and also output of `bin\install.ps1` likes `Initializing...` and `Downloading scoop...`...

**Comment by maphew** [comment]
@niheaven it seems there is no output for `Set-ExecutionPolicy RemoteSigned -scope CurrentUser` when it's in a script, at least that's what happens for my machine.

~~~
PS C:\temp> echo "Set-ExecutionPolicy RemoteSigned -scope CurrentUser" > x.ps1
PS C:\temp> .\x.ps1
PS C:\temp>
~~~

Here's the complete script and it's output as it stands now
https://gist.github.com/maphew/8b9ad8ba64c7121cd7502bfcef8ce025

I think the "Initializing..." etc. output might have been missing because when cleaning the environment to try again I was not removing `$userprofile\.config\scoop` (which I didn't know about until today).

**Reply by maphew** [reply]
I created an issue https://github.com/ScoopInstaller/Scoop/issues/4628
